glommio/src/executor.rs

```diff
@@ -1050,32 +1050,9 @@ impl<T> Task<T> {
             }
         });
 
-        if !need_yield {
-            return;
+        if need_yield {
+            futures_lite::future::yield_now().await;
```

Comment:
thank you!! When I found out about `yield_now` I figured it would do the same as this, and put it on my backburner to go back and verify.

Never did, though.